meetings/working-groups/collection-literals/LDM-questions-2023-08-15.md

```diff
@@ -0,0 +1,223 @@
+
+    Spans are better than interfaces both because they're already better than arrays, and because they can be thought of as morally implementing those interfaces (which may also be literally true in the future).
+
+    However, this has oddities:
+
+    ```c#
+    void M(List<int> list);
+    void M(ReadOnlySpan<int> list);
+
+    M([1, 2, 3]); // Ambiguous.
+    ```
+
+    This seems odd.  If there are collection overloads, and one is concrete, and one is a span, we would say it was ambiguous, when it seems clear that one exists for perf and should be preferred.
+
+1. Option B: Same as option A, except that it applies to all ref-structs, not just Span/ReadOnlySpan.
+
+    The general intuition is that anything in the ref-struct realm should be considered fast/span-like and then fit into this bucket.  
+
+    With this approach the following would also hold:
+
+    ```c#
+    ref struct ValueList<T> { ... }
+
+    void M(T[] list);
+    void M(ValueList<int> list);
+
+    M([1, 2, 3]); // calls the ValueList version.
+    ```
+
+    However, like Option A, this would become ambiguous with other concrete overloads:
+
+    ```c#
+    ref struct ValueList<T> { ... }
+
+    void M(List<int> list);
+    void M(ValueList<int> list);
+
+    M([1, 2, 3]); // Ambiguous.
+    ```
+
+1. Option C: Ref-structs are always considered better than all other types.
+
+    The general intuition here is that if the user provides a ref-struct overload that that is the 'fast' option, and thus would be preferred over other types.  This seems like the 'more likely than not' case that extends beyond just Span/ReadOnlySpan. Also seems to align with things like interpolated string handlers over System.String.
+
+    For example:
+
+    ```c#
+    ref struct ValueSet<T> { ... }
+
+    void M(ValueSet<int> list);
+    void M(HashSet<int> list);
+
+    M([x, y, z]); // calls the ValueList version.
+    ```
+
+    The general intuition here is that if the user provides a ref-struct overload that that is the 'fast' option, and thus would be preferred over other types.  This seems like the 'more likely than not' case that extends beyond just Span/ReadOnlySpan.
+
+WG opinion: Leaning towards Option C, prefer ref-structs over all other types.
+
+### Add support for `Memory<T>`
+
+Should we add support for `Memory<T>` in C# 12.  For example:
+
+```c#
+Memory<int> m = [1, 2, 3];
+```
+
+Based on last LDM meeting, we are deciding "We will do if we have time, but it is low pri and can be cut if we do not have the time/resources"
+
+1. We made a similar decision for inline-array types.  These types are not in the core majority of apis/use-cases, and can come later.
+
+1. Adding support for a new type *is* a breaking change.  However, this is unavoidable in this space as any type can become a collection type in the future (through usage of the `[CollectionBuilder]` attribute).  As such, we do not think it critical to front load support to avoid this problem.
```

Comment:
Adding a `[CollectionBuilder]` builder attribute to a core type would require analyzing the breaking change impact as well. Perhaps just combine this bullet with the following, and remove the attribute example.

```suggestion
1. Adding support for a new type *is* a breaking change.  However, this is unavoidable in this space as any type can become a collection type in the future (through usage of the `[CollectionBuilder]` attribute).  As such, we do not think it critical to front load support to avoid this problem.
```